First draft for en/getting-started.md (#297)
JabRef/user-documentation: en/getting-started.md

# Getting started

## Main Window of JabRef

Upon the first start of JabRef the main user interface is showing up the main elements are:

- Menu bar
- Icon bar (shortcuts for most frequently used features)
- Side bar (for groups and web search)

![Screenshot of main window](.gitbook/assets/getting-started-main-screen.png)

## Creation of a new library

> A "library" is the main file that saves all the information about your collection of references. The storage format of the file is text-based in the Bibtex standard.

Reviewer: I would move this information about bibtex to https://docs.jabref.org/cite/bibtex-and-biblatex. It's not utterly important for a new user how JabRef saves his information (except when he starts to cite papers in tex).

Author: As this is for me still one of the essential aspects of JabRef I would keep it ;-)

The key usually follows a global pattern and can be easily created automatically by clicking on the "generate" button next to the field.

{% hint style="info" %}
The default key pattern is `[auth][year]`, which means that Author information is followed by the year of the publication, resulting in the example in `Turing1950`. However, the key pattern is customizable to your needs. See [Configuration](https://docs.jabref.org/setup) > ["Customize the citation key generator"](https://docs.jabref.org/setup/bibtexkeypatterns) for more details.
{% endhint %}

After entering some information, you can see on the right side of the entry editor a preview of the bibliographic data:

![Added information for new entry](.gitbook/assets/getting-started-filled-entry-editor.png)

> 

Reviewer: Point them to https://docs.jabref.org/collect for further possibilities to add entries.

Author: Done - hopefully in the right syntax - needs to be confirmed after pushing to master in "Gitbook" view.

## Enhancing the information

After creating the basic information the addition of all other bibliographical details is often cumbersome and error-prone. To ease this task, JabRef allows for an automatic completion of the bibliographic information by looking up the data in public databases.
To use this feature just click on the "Update with bibliographic information from the web" button in the editor:

![Update information from web](.gitbook/assets/getting-started-entry-editor-update-from-web.png)

{% hint style="info" %}

> The found information is most accurate if an identifier like a "DOI" or "ISBN" is maintained. If you already know such an unique identifier, this can also be already the starting point to create a new entry without manual entering any information by using the "create from ID" feature in the Create entry dialog. For more information see: [Collect](https://docs.jabref.org/collect) > ["Add entry using an ID"](https://docs.jabref.org/collect/add-entry-using-an-id)

Reviewer 2: Maybe add a hint that you can paste DOIs direclty on the main table?

{% endhint %}

If additional information is found you will be asked in a dialog which information should be taken over:

![Merging the existing and the web information](.gitbook/assets/getting-started-merge-entries.png)

## Adding a full text document

> Usually you also want to attach a reference to the full-text of a reference. For this, you can use the "file" field in the "General" tab.

Reviewer 2: ```suggestion
Usually, you also want to attach a reference to the full-text of a reference. For this, you can use the "file" field in the "General" tab.
```

Author: Done.

Here you can either attach a file manually, search for an already existing local file matching the bibtexkey pattern, or trying to automatically download a matching full text from the web.

{% hint style="info" %}

> In order to use the automated feature, it is necessary to set-up a file directory first. To do so, please go to "Options" > "Preferences", go to "File" section and select there an existing folder as the "Main file directory":

Reviewer 2: ```suggestion
In order to use the automated feature, it is necessary to set-up a file directory.  The file directory can either be global, local to a user, or specific for a library. See the help on [Override file directories](https://docs.jabref.org/setup/databaseproperties#override-default-file-directories) for details. 
For beginners, a main file directory is the most simple setup. It can be set in "Options" -> "Preferences" -> "File"  
```

Reviewer: The default behavior is that JabRef looks for files relative to the bib file, so I would describe this in the getting started section, an then point them to the "Override file directories" section if they want to change that.

If we need to point users to the preferences in "getting started", then we should change the default preferences in my opinion.

Author: Unfortunately, this is not the case - I actually reset the settings before creating the screenshots - and without and change in the settings the following dialog appears:
![grafik](https://user-images.githubusercontent.com/676652/92331417-b57e9380-f076-11ea-8032-1c0d135decfa.png)

Default Prefs setting: 
![Grabbed_20200906-192612](https://user-images.githubusercontent.com/676652/92331434-e232ab00-f076-11ea-9b7e-f0ff5f5e0018.png)

Reviewer 2: Yep. That's a bug which was somehow introduced in 5.1. No idea how it came.

![Setting up the main file directory](.gitbook/assets/getting-started-preferences.png)
{% endhint %}

> To test the automatic download of full texts you can click on the "Get full-text" icon next to the file field, or choose "Lookup" -> "Search full text documents online" from the menu. As soon as a full-text is found, the file will be stored in the local file directory and linked to the entry:

Reviewer 2: ```suggestion
To test the automatic download of full texts you can click on the "Get full-text" icon next to the file field, or choose "Lookup" -> "Search full text documents online" from the menu. As soon as a full-text is found, the file will be stored in the previous defined file directory and linked to the entry:
```

![Finding a full-text document online](.gitbook/assets/getting-started-entry-editor-full-text.png)

> To open the downloaded full text you can click on "file" icon before the file name - or use the same icon in the entry table:

Reviewer 2: ```suggestion
To open the downloaded full text you can click on the "file" icon before the file name - or use the same icon in the entry table:
```

Author: Done.

![Opening the full-text](.gitbook/assets/getting-started-open-fulltext.png)

## Finding more references in the web

> If you want to search for other references, it is also possible to directly trigger a search in many of the most common bibliographic databases. To start a search just us the "Web Search" feature of JabRef:

Reviewer 2: ```suggestion
If you want to search for other references, it is also possible to directly trigger a search in many of the most common bibliographic databases. To start a search just use the "Web Search" feature of JabRef:
```

Author: done.

First select one of the existing data sources, enter a search term and click on "search":

> The search results will be shown in an window where you can select all the search hits to be added to your library.

Reviewer 2: ```suggestion
The search results will be shown in an window where you can select all the search hits to be added to your library.
You also have the option to directly download the fulltext document (if avaiable and possible).
```

Reviewer: That's not relevant for a getting started in my opinion

Reviewer 2: I was referring to the dialog, that it has an option to download the ffull text .

![Web Search: Trigger and result window](.gitbook/assets/getting-started-import-from-web.png)

## Organize your library

> {% hint style="info" %}

Reviewer: I would simply point the user to the organizing section for more information about grouping etc. This is not that important if you start JabRef for the first time in your life.

Moreover, I would add a link to https://docs.jabref.org/cite, so that the user has all the information for a complete research > write workflow.

Author: Done.
